stdlib/builtins.pyi

```diff
@@ -1619,12 +1618,27 @@ class reversed(Iterator[_T], Generic[_T]):
     def __length_hint__(self) -> int: ...
 
 def repr(__obj: object) -> str: ...
+
+# See https://github.com/python/typeshed/pull/9141 on why we don't use
+# `SupportsRound` from `typing.pyi`
+
+class _SupportsRound1(Protocol[_T_co]):
+    def __round__(self) -> _T_co: ...
+
+class _SupportsRound2(Protocol[_T_co]):
+    def __round__(self, __ndigits: int) -> _T_co: ...
+
+# We need these overloads to replicate how stdlib works
+@overload
+def round(number: float, ndigits: None = ...) -> int: ...
```

Comment:
I haven't looked at the code deeply, but I feel like we shouldn't need these if `float.__round__` is defined correctly in our stubs. I believe the C code for `builtins.round` doesn't actually treat `float` specially.